# 1. The World Bank's International Debt Data

It's not just individuals who incur debt to meet their needs; countries also resort to borrowing to manage their economies. For instance, significant investments in infrastructure are essential for enhancing the quality of life for citizens. The World Bank is a prominent institution that extends loans to nations for such purposes.

The dataset provided here offers insights into international debt by capturing information about the amount of debt owed by developing countries across different categories. Collected and maintained by The World Bank, this dataset spans several decades, __covering debt statistics from 1970 to 2015 for numerous countries worldwide.__

In this notebook, we will delve into the analysis of international debt data compiled by The World Bank. The dataset encompasses details regarding the indebtedness (in USD) of developing nations across various categories. Through our exploration, we aim to uncover insights into queries such as:

__What is the total amount of debt that is owed by the countries listed in the dataset?__

__Which country owns the maximum amount of debt and what does that amount look like?__

__What is the average amount of debt owed by countries across different debt indicators?__


Let's first SELECT all of the columns from the international_debt table. Also, we'll limit the output to the first ten rows to keep the output clean.

In [3]:
%%sql

SELECT * FROM international_debt Limit 10;

 * mysql+mysqlconnector://root:***@localhost:3306/debtanalysis
10 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.70000')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.10000')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.90000')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.40000')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.10000')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', Decimal('39107845.00000')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', Decimal('23779724.30000')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', Decimal('13335820.00000')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', Decimal('100847181.90000')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', Decimal('72894453.70000'))]

# 2. Determining the count of distinct countries

Upon reviewing the initial ten rows, we observe Afghanistan's debt across various debt indicators. However, the repetition of country names suggests multiple instances of debt within the dataset. To ensure a comprehensive statistical analysis, it's imperative to ascertain the total count of distinct countries represented. Hence, __in this section, we aim to extract the number of unique countries recorded in the table.__

In [4]:
%%sql

SELECT COUNT(DISTINCT(country_name)) AS Total_distinct_countries
FROM international_debt;

 * mysql+mysqlconnector://root:***@localhost:3306/debtanalysis
1 rows affected.


[(124,)]

# 3. Determing the distinct debt indicators

We observe that the table comprises a total of 124 distinct countries. As discussed earlier, the 'indicator_name' column provides a succinct description of the debt's purpose. Adjacent to it, the 'indicator_code' column denotes the category to which these debts belong. __Familiarizing ourselves with these diverse debt indicators__ facilitates comprehension of the potential areas for a country's indebtedness.

In [5]:
%%sql

SELECT DISTINCT(indicator_code) as distinct_debt_indicators
FROM international_debt
ORDER BY distinct_debt_indicators

 * mysql+mysqlconnector://root:***@localhost:3306/debtanalysis
25 rows affected.


[('DT.AMT.BLAT.CD',),
 ('DT.AMT.DLXF.CD',),
 ('DT.AMT.DPNG.CD',),
 ('DT.AMT.MLAT.CD',),
 ('DT.AMT.OFFT.CD',),
 ('DT.AMT.PBND.CD',),
 ('DT.AMT.PCBK.CD',),
 ('DT.AMT.PROP.CD',),
 ('DT.AMT.PRVT.CD',),
 ('DT.DIS.BLAT.CD',),
 ('DT.DIS.DLXF.CD',),
 ('DT.DIS.MLAT.CD',),
 ('DT.DIS.OFFT.CD',),
 ('DT.DIS.PCBK.CD',),
 ('DT.DIS.PROP.CD',),
 ('DT.DIS.PRVT.CD',),
 ('DT.INT.BLAT.CD',),
 ('DT.INT.DLXF.CD',),
 ('DT.INT.DPNG.CD',),
 ('DT.INT.MLAT.CD',),
 ('DT.INT.OFFT.CD',),
 ('DT.INT.PBND.CD',),
 ('DT.INT.PCBK.CD',),
 ('DT.INT.PROP.CD',),
 ('DT.INT.PRVT.CD',)]

# 4. Calculating the aggregate debt owed by all countries

As previously discussed, the financial indebtedness of individual countries mirrors their economic condition. However, when considering this on a global scale, how should we tackle it?

Now, shifting our focus from debt indicators, __let's calculate the aggregate debt amount (in USD) owed by various countries.__ This will provide insight into the overall state of the global economy.

In [6]:
%%sql

SELECT ROUND(SUM(debt),2) as total_debt
FROM international_debt; 

 * mysql+mysqlconnector://root:***@localhost:3306/debtanalysis
1 rows affected.


[(Decimal('3079734487675.80'),)]

__Human perception struggles with comprehending extremely large or small numbers__, as Daniel Kahneman suggests. For instance, more than 3 trillion USD is a figure that surpasses our cognitive grasp.

In [7]:
%%sql

SELECT ROUND(SUM(debt)/1000000,2) as total_debt
FROM international_debt; 

 * mysql+mysqlconnector://root:***@localhost:3306/debtanalysis
1 rows affected.


[(Decimal('3079734.49'),)]

# 5. Nation with most significant debt

Now that we've obtained the precise total of debt amounts owed by various countries, __let's identify the nation with the highest debt and its corresponding amount.__ It's important to note that this debt encompasses various categories, providing insights into the socio-economic landscape of the country. Although we could also determine the category in which the country holds its highest debt, we'll defer that analysis for now."

In [8]:
%%sql

SELECT 
    country_name, 
    ROUND(SUM(debt),2) as total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt desc
LIMIT 1;

 * mysql+mysqlconnector://root:***@localhost:3306/debtanalysis
1 rows affected.


[('China', Decimal('285793494734.20'))]

# 6. Average amount of debt across indicators

So, _China_ emerged as the country with the highest debt.

Having obtained a concise overview of the dataset and some of its summary statistics, we now understand the various debt indicators under which countries accumulate their debts. To gain deeper insights, we can explore the average debt owed by each country. This will provide us with a clearer understanding of how debt amounts are distributed across different indicators.

In [9]:
%%sql
SELECT 
    indicator_code AS debt_indicator,
    indicator_name,
    AVG(debt) as average_debt
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY average_debt desc
LIMIT 10;

 * mysql+mysqlconnector://root:***@localhost:3306/debtanalysis
10 rows affected.


[('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('5904868401.499193629')),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('5161194333.812658227')),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', Decimal('2152041216.890243902')),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('1958983452.859836065')),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', Decimal('1803694101.963265306')),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', Decimal('1644024067.650806451')),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', Decimal('1223139290.398230088')),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', Decimal('1220410844.421518987')),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('1191187963.083064516')),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', Decimal('1082623947.653623188'))]

# 7. The maximum sum of principal repayments

Observing the data, it's evident that the __indicator DT.AMT.DLXF.CD stands out as the leader in average debt.__ This category primarily encompasses the repayment of long-term debts, which countries typically undertake to secure immediate capital. For further insights into this category, additional information can be explored.

A noteworthy observation from the aforementioned finding is the substantial disparity in the amounts of indicators following the second one. This suggests that the first two indicators might represent the most significant categories in which countries are indebted.

Further investigation can shed light on _which country carries the highest debt burden in the long-term debt category (DT.AMT.DLXF.CD)_. Given the diverse economic challenges faced by different nations, this discovery will provide deeper insights into the economic condition of that specific country.

In [10]:
%%sql

SELECT country_name, debt, indicator_name, indicator_code
FROM international_debt
WHERE indicator_code = 'DT.AMT.DLXF.CD'
ORDER BY debt DESC
LIMIT 10;

 * mysql+mysqlconnector://root:***@localhost:3306/debtanalysis
10 rows affected.


[('China', Decimal('96218620835.70000'), 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD'),
 ('Brazil', Decimal('90041840304.10001'), 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD'),
 ('Russian Federation', Decimal('66589761833.50000'), 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD'),
 ('Turkey', Decimal('51555031005.80000'), 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD'),
 ('South Asia', Decimal('48756295898.20000'), 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD'),
 ('India', Decimal('31923507000.80000'), 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD'),
 ('Indonesia', Decimal('30916112653.80000'), 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD'),
 ('Kazakhstan', Decimal('27482093686.40000'), 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD'),
 ('Mexico', Decimal('25218503927.00000'), 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD'),
 ('Least developed countries: UN classification', Decimal('25197029299.40000'), 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD')]

# 8. The most frequently occurring debt indicator

_China holds the highest amount of debt in the long-term debt (DT.AMT.DLXF.CD) category, as confirmed by The World Bank._ It's prudent to validate our analyses to ensure accuracy and reliability.

While long-term debt emerges as the leading category in terms of average debt amount, is it also the most prevalent debt indicator among countries? Let's explore further to ascertain this.

In [11]:
%%sql
SELECT 
    indicator_code,
    COUNT(indicator_code) as indicator_count
FROM international_debt
GROUP BY indicator_code
ORDER BY indicator_count desc, indicator_code desc
LIMIT 20;

 * mysql+mysqlconnector://root:***@localhost:3306/debtanalysis
20 rows affected.


[('DT.INT.OFFT.CD', 124),
 ('DT.INT.MLAT.CD', 124),
 ('DT.INT.DLXF.CD', 124),
 ('DT.AMT.OFFT.CD', 124),
 ('DT.AMT.MLAT.CD', 124),
 ('DT.AMT.DLXF.CD', 124),
 ('DT.DIS.DLXF.CD', 123),
 ('DT.INT.BLAT.CD', 122),
 ('DT.DIS.OFFT.CD', 122),
 ('DT.AMT.BLAT.CD', 122),
 ('DT.DIS.MLAT.CD', 120),
 ('DT.DIS.BLAT.CD', 113),
 ('DT.INT.PRVT.CD', 98),
 ('DT.AMT.PRVT.CD', 98),
 ('DT.INT.PCBK.CD', 84),
 ('DT.AMT.PCBK.CD', 84),
 ('DT.INT.DPNG.CD', 79),
 ('DT.AMT.DPNG.CD', 79),
 ('DT.INT.PBND.CD', 69),
 ('DT.AMT.PBND.CD', 69)]

# 9. Other relevant debt matters and final thoughts

In our dataset, all listed countries have incurred debt in a total of six distinct indicators. Notably, the indicator DT.AMT.DLXF.CD is among them, hinting at a common economic challenge shared by these nations. However, this merely marks a pivotal moment in our exploration rather than its conclusion.

Transitioning our focus from debt indicators, _let's revisit the analysis of debt amounts. We'll identify the maximum debt amount across various indicators and its corresponding country._ This endeavor will shed light on additional potential economic issues faced by countries. By the conclusion of this section, we'll discern the primary debt indicators associated with a country's highest debt.

Throughout this notebook, we've examined global debt obligations of countries, extracting key summary statistics and uncovering intriguing insights. Our findings have been rigorously validated to ensure the accuracy of our investigations.

In [12]:
%%sql
select
country_name,
indicator_code,
MAX(debt) as maximum_debt
from international_debt
GROUP BY country_name, indicator_code
ORDER BY maximum_debt desc
LIMIT 10;

 * mysql+mysqlconnector://root:***@localhost:3306/debtanalysis
10 rows affected.


[('China', 'DT.AMT.DLXF.CD', Decimal('96218620835.70000')),
 ('Brazil', 'DT.AMT.DLXF.CD', Decimal('90041840304.10001')),
 ('China', 'DT.AMT.DPNG.CD', Decimal('72392986213.80000')),
 ('Russian Federation', 'DT.AMT.DLXF.CD', Decimal('66589761833.50000')),
 ('Turkey', 'DT.AMT.DLXF.CD', Decimal('51555031005.80000')),
 ('South Asia', 'DT.AMT.DLXF.CD', Decimal('48756295898.20000')),
 ('Brazil', 'DT.AMT.PRVT.CD', Decimal('43598697498.60000')),
 ('Russian Federation', 'DT.AMT.DPNG.CD', Decimal('42800154974.90000')),
 ('Brazil', 'DT.AMT.DPNG.CD', Decimal('41831444053.30000')),
 ('Least developed countries: UN classification', 'DT.DIS.DLXF.CD', Decimal('40160766261.60000'))]